src/zenml/integrations/pandas/materializers/pandas_materializer.py

```diff
@@ -325,3 +325,42 @@ def extract_metadata(
                 }
 
         return pandas_metadata
+
+    def get_item_count(self, data: Any) -> Optional[int]:
+        """Get the number of items for the given data.
+
+        Args:
+            data: The data to get the number of items for.
+
+        Returns:
+            The number of items for the given data.
+        """
+        if isinstance(data, pd.Series):
+            return len(data)
+        elif isinstance(data, pd.DataFrame):
+            return int(data.shape[1])
+        return None
```

Comment:
My immediate thought to this is that we probably need to be more formal about sequence-typed artifacts and also store the artifact element data type, in addition to the count. We would need this to validate and/or return the correct data type when unpacking elements in 'load_chunk'

Reply:
I thought about this as well, but what would the element type of a DataFrame even be? Somehow a tuple of all the column types?